File: glumacplusandroid/app/src/main/java/com/example/glumac_plus_android/data/model/Proizvod.kt
package com.example.glumac_plus_android.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Proizvod(
    val id: Long,
    val naziv: String,
    val opis: String? = null,
    val cena: Double,                 
    val tip: String,
    val alergeniNazivi: List<String>? = null
)

data class StavkaKorpe(
    val proizvod: Proizvod,
    val kolicina: Int
)

File: frontend/src/types/proizvod.ts
// Centralni domenski tip proizvoda — koristi se na meniju, u korpi, na kartici i u omiljenima.
export interface Proizvod {
    id: number;
    naziv: string;
    opis?: string;
    cena: number;
    tip: string;
    alergeniNazivi?: string[];
}

// Stavka u korpi = proizvod + količina.
export interface StavkaKorpe {
    proizvod: Proizvod;
    kolicina: number;
}
